packages/x-data-grid-premium/src/components/GridGroupingColumnLeafCell.tsx

```diff
@@ -9,13 +9,16 @@ function GridGroupingColumnLeafCell(props: GridRenderCellParams) {
 
   return (
     <Box
-      sx={{
-        ml:
-          rootProps.rowGroupingColumnMode === 'multiple'
-            ? 1
-            : (theme) =>
+      sx={[
+        rootProps.rowGroupingColumnMode === 'multiple'
+          ? {
+              ml: 1,
+            }
+          : {
+              ml: (theme) =>
```

Comment:
Should use the `sx` callback signature instead?

Reply:
@siriwatknp confirmed that this result from codemod is correct. 🤔

Reply:
Isn't it the opposite of https://github.com/mui/mui-x/pull/14143#discussion_r1718271042?
Or am I missing something?

Reply:
This looked strange to me as well. 🙈 
I'm asking @siriwatknp for a review of this PR to check these changes. 🤔

Reply:
Sorry, I missed one case. It should be:

```js
sx={[
  rootProps.rowGroupingColumnMode === 'multiple'
          ? {
              ml: 1,
            }
          : (theme) => ({
              ml: `calc(var(--DataGrid-cellOffsetMultiplier) * ${theme.spacing(rowNode.depth)})`
            })
]}
```

Reply:
> This looked strange to me as well

It looks strange to me at first but if you view it from static perspective, it starts to make sense.

Each style (an object of CSS) must be static, so callback as values are not allowed. However, each style can be a callback in order to access the theme (theme is static because it's configured at build time). So here is what style can be:

1. plain object: `{ display: 'flex', ml: 2, …etc }`
2. a callback that return an object: `theme => ({ display: 'flex', ml: theme.spacing(2), …etc })`

Each style generates a class name that will be used as a reference to the style.

Now the dynamic part. From static point of view, you cannot switch the value of a CSS property based on condition but what you can do is switching classes (styles) based on condition. That's why the result looks like:

`sx={condition ? { … } : { … }}` => transformed to `sx={condition ? 'classname1' : 'classname2'}`

The same concept applies for multiple styles using array:

```js
sx={[
  condition1 ? { … } : (theme => {…}),
  condition2 ? { … } : { … }
]}
```

Does this help?

Reply:
@siriwatknp what about this:

```js
      sx={
        (theme) => ({
           ml: rootProps.rowGroupingColumnMode === 'multiple'
             ? 1
             : `calc(var(--DataGrid-cellOffsetMultiplier) * ${theme.spacing(rowNode.depth)})`
         })
       }
```

- Does this work?
- If it works, is there a preferred approach?

Reply:
Alright, I was confused by this: `ml: (theme) => {...}` 😅

The theme is only available when using the callback signature on the **style object level** 👍🏻